src/sentry/utils/sdk.py

```diff
@@ -445,11 +454,41 @@ def _kwargs_into_scope(self, scope, extra=None, tags=None, fingerprint=None, req
             scope.fingerprint = fingerprint
 
 
+def check_tag(tag_key: str, expected_value: str) -> bool:
+    """Detect a tag already set and being different than what we expect.
+
+    This function checks if a tag has been already been set and if it differs
+    from what we want to set it to.
+    """
+    with configure_scope() as scope:
+        if scope._tags and tag_key in scope._tags and scope._tags[tag_key] != expected_value:
```

Comment:
The main difference with #45039 is that we also check for the tags being different ( `scope._tags[tag_key] != expected_value`).